File: java/lists_2/Main.java
import java.io.FileOutputStream;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;

public class Main {
    public static void main(String[] args) {

        List<Integer> numbers = generateIntegersList();
        List<String> words = List.of("RAZ", "DWA", "TRZY", "CZTERY", "PIEC", "SZESC", "SIEDEM", "OSIEM", "DZIEIWEC");
        List<Integer> numbersPrime = new ArrayList<>(new ArrayList<>(numbers
                .stream()
                .filter(num -> isPrime(num))
                .collect(Collectors.toSet())));
        List<String> numbersPrimeTranslated = transformNumbersToStrings(numbersPrime, words);
        saveDataToFile(numbersPrimeTranslated, "output");

        System.out.println("Wszystkie liczby: " + numbers);
        System.out.println("Liczby pierwsze: " + numbersPrime);
        System.out.println("Liczby w zapisie słownym: " + numbersPrimeTranslated);
    }

    private static Integer generateNumber(int from, int to) {
        return (int) Math.floor(Math.random() * (to - from) + from);
    }

    private static List<Integer> generateIntegersList() {
        List<Integer> output = new ArrayList<>();
        int howManyIntegers = generateNumber(10, 100);
        for (int i = 0; i < howManyIntegers; i++) {
            output.add(generateNumber(10, 100));
        }
        return output;
    }

    private static boolean isPrime(int number) {
        if (number < 2) {
            return false;
        }
        for (int i = 2; i < number; i++) {
            if (number % i == 0) {
                return false;
            }
        }
        return true;
    }

    private static int findBiggerDigit(int number) {
        return Arrays.stream(String.valueOf(number).split(""))
                .map(Integer::parseInt)
                .reduce((n1, n2) -> n1 > n2 ? n1 : n2)
                .get();
    }

    private static List<String> transformNumbersToStrings(List<Integer> numbers, List<String> words) {
        List<String> output = new ArrayList<>();

        for (int i = 0; i < numbers.size(); i++) {
            output.add(words.get(findBiggerDigit(numbers.get(i)) - 1));
        }
        return output;
    }

    private static void saveDataToFile(List<String> data, String fileName) {
        StringBuilder stringBuilder = new StringBuilder();
        data
                .forEach(word -> stringBuilder.append(word).append(","));
        try {
            FileOutputStream stream = new FileOutputStream(System.getProperty("user.dir") + "/" + fileName);
            byte[] bytes = stringBuilder.toString().getBytes(StandardCharsets.UTF_8);
            stream.write(bytes);
            stream.close();

        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}

File: python/lists_2/main.py
import random
import math


def generate_number(min_num, max_num):
    return random.randint(min_num, max_num)


def generate_array():
    return [generate_number(10, 100) for x in range(generate_number(10, 100))]


def is_prime(number):
    for i in range(2, number):
        if number % i == 0:
            return False
    return True


def get_primes_only(input_array):
    return [x for x in input_array if is_prime(x)]


def write_bigger_digit_as_text(input_array):
    output = []

    for number in input_array:
        first_digit = math.floor(number / 10)
        second_digit = number % 10
        bigger_digit = first_digit if first_digit > second_digit else second_digit

        if bigger_digit == 1:
            output.append('JEDEN')
        elif bigger_digit == 2:
            output.append('DWA')
        elif bigger_digit == 3:
            output.append('TRZY')
        elif bigger_digit == 4:
            output.append('CZTERY')
        elif bigger_digit == 5:
            output.append('PIEC')
        elif bigger_digit == 6:
            output.append('SZESC')
        elif bigger_digit == 7:
            output.append('SIEDEM')
        elif bigger_digit == 8:
            output.append('OSIEM')
        elif bigger_digit == 9:
            output.append('DZIEWIEC')
        else:
            output.append('ZERO')
    return output


array = generate_array()
prime_array = get_primes_only(array)
text_array = write_bigger_digit_as_text(prime_array)

print(f'Wylosowano tablicę liczb: {array}')
print(f'Tablica tylko z liczbami pierwszymi: {prime_array}')
print(f'Tablica z zapisem większej cyfry jako tekstu {text_array}')

file = open('sentences.txt', 'w')
file.write(str(text_array))
file.close()
